### DISCUSSION TITLE
Use number of nodes in a nodeset in an equation

### DISCUSSION THREAD

**Discussion by Edward-Eth** [idaholab/moose#21463] [Q&A General] [answered]
Hi, I am attempting to apply an out of plane distributed load to a shell element model. Looking at examples, and with experiments of my own, it seems the ADPressure BC doesn't allow the application of loads in the Z direction on a boundary on a 2D X-Y generated mesh. The example `(modules/tensor_mechanics/test/tests/shell/static/beam_bending_moment_AD.i)` shows the use of a ConstantRate nodal kernel to apply an out of plane distributed nodal force, however if you don't know the number of nodes on the boundary while writing the input file, it is impossible to correctly define a distributed load this way. As such I was wondering if it is possible to return the number of nodes in a boundary to a function to calculate a force per node equivalent to a certain desired "total" force (ie pressure x area)

**Comment by GiudGiud** [comment]
Hello

In a function directly no, but you can use a postprocessor to count the nodes, then use a parsedfunction to use that value in a function. 

https://mooseframework.inl.gov/source/postprocessors/NumNodes.html
https://mooseframework.inl.gov/source/functions/MooseParsedFunction.html

Guillaume

**Reply by Edward-Eth** [reply]
Is NumNodes able to operate on a subdomain or boundary to return the number of nodes in/on the same? It seems to lack a boundary or block input to allow it to do that?

**Reply by GiudGiud** [reply]
It cannot. We'd need to add block or boundary restriction.
It's not very hard to do, do you need block or boundary?

**Comment by KhaledNabilSharafeldin** [comment, marked as answer]
If these nodes belong to a sideset, and you know the sideset name/ID, you can get the list of node IDs in that sideset and apply the condition in an Action.

so all you need to do is. make a new action that has a function that sets up the list of node IDs, then take that list and use it as a parameter to set up the BC you're trying to do. 

I am not sure which BCs you are trying to use but that much I know.
you can get the boundaryID by name using `_mesh->getBoundaryID(name)`
and you can use that ID to get the list of nodeIDs `_mesh->getNodeList(boundaryID)` which returns a `std::vector<dof_id_type>`

**Reply by GiudGiud** [reply]
yup from the code-side you can use these routines.
